### DISCUSSION TITLE
Allow Expo/React-Native apps to call NativeScript to handle low level operations that would've required platform level code.

### DISCUSSION THREAD

**Discussion by trajano** [NativeScript/NativeScript#10635] [Ideas]
I was playing around with the Vibration feature of React-native to play Shave and a Haircut
```typescript
        <VibrateButton title="Shave and a Haircut" pattern={[
          0, // no gap
          200, // shave
          400,
          100, // and
          100,
          100, // a
          200,
          400, // hair
          200,
          300, // cut
          1000, // long pause
          400, // two
          200,
          400 // bits
        ]} />
```
works well on Android, but doesn't work on iOS.

So I did a quick research and there is likely a way of doing it using `CHHapticPattern.advancedPlayer`.

That got me to thinking is there a way of getting around of switching to the macOS and I recall NativeScript can do native calls.  So I was thinking what if Expo just had a bridge to call NativeScript somehow, I presume it would be a config-plugin and likely needs to be "compiled" separately

**Comment by shirakaba** [comment]
I made a proof of concept of calling NativeScript from an Expo app. Here's an Expo [template](https://github.com/shirakaba/jamie-expo-templates/tree/nativescript/templates/nativescript) for it.

Unfortunately, it's limited to Expo SDK 50, React Native v0.72.4 for now (as that's the newest version of React Native that our fork of Hermes Windows works with). I won't be able to update these forks for newer versions of React Native and Expo as it's beyond my expertise. However, I'm hoping in the next year that Node-API support will [land in Hermes core](https://github.com/facebook/hermes/pull/1377) and we'll no longer need any of this boilerplate to make this possible.

**Comment by NathanWalker** [comment]
Using what @shirakaba describes, I was able to use it with Expo as well; worked nicely:
https://x.com/wwwalkerrun/status/1783503392771547474?s=46&t=xO7ZRuPOKvNVkUfT7PjvTQ

Looking forward to the Hermes PR linked above landing to help make it broadly available.
